src/app/search-results/search-results.component.ts

```diff
@@ -0,0 +1,68 @@
+import { Component, OnDestroy, OnChanges, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Channel } from '../shared/models/channel.model';
+import { Video } from '../shared/models/video.model';
+import { ChannelService } from '../core/services/channel.service';
+import { VideoService } from '../core/services/video.service';
+
+@Component({
+  selector: 'bs-search-results',
+  templateUrl: './search-results.component.html',
+  styleUrls: ['./search-results.component.scss']
+})
+export class SearchResultsComponent implements OnDestroy, OnInit {
+
+  constructor(
+    private route: ActivatedRoute,
+    private channelService: ChannelService,
+    private videoService: VideoService
+    ) { }
+
+  routeQuerySub: any;
+  videosSub: any;
+  channelsSub: any;
+  channels: Channel[];
+  videos: Video[];
+  search: string;
+  videosToLoad = 40;
+  channelsToLoad = 40;
+
+  ngOnInit() {
+    this.routeQuerySub = this.route.queryParams
+    .subscribe( params => {
+      this.channels = [];
```

Comment:
You don't have to reassign `this.channels` and `this.videos` to `[]` if you are loading video and channels later in the same function

Reply:
i have to init them every query-string change. 
otherwise the previous search results will stay in the next search.
the route render once, and only the routeQuerySubscription change each queryParams chane.